internals/overlord/checkstate/manager.go

```diff
@@ -14,30 +14,57 @@
 
 package checkstate
 
+// TODO: should it only go to recover-check when it hits the threshold?
+
 import (
 	"context"
 	"sort"
 	"sync"
-	"time"
+	"sync/atomic"
 
-	"github.com/canonical/pebble/internals/logger"
+	"github.com/canonical/pebble/internals/overlord/state"
 	"github.com/canonical/pebble/internals/plan"
 )
 
+const (
+	noPruneAttr = "checkstate-no-prune"
+)
+
 // CheckManager starts and manages the health checks.
 type CheckManager struct {
-	mutex           sync.Mutex
-	wg              sync.WaitGroup
-	checks          map[string]*checkData
+	state      *state.State
+	ensureDone atomic.Bool
+
+	checksLock sync.Mutex
+	checks     map[string]*checkData
```

Comment:
Super nitty - I saw when we are iterating over ```checks``` the item is called ```data``` which was not super clear when I read it, so I came here, and see the item is called ```checkData```. Is there a reason why ```check``` did not work ? some namespace clash ?